bgp/modules/terms.py

```diff
@@ -113,3 +115,55 @@ def validate_isbn(isbn):
     def __init__(self):
         super(IsbnExtractorModule, self).__init__(self.validate_isbn)
 
+
+class PageTypeProcessor:
+
+    def __init__(self, modules):
+        self.modules = modules
+
+    '''
+    1 --> run() method sends the book(djvu.xml) file to pageProcessor.
+    2 --> pageProcessor processes the xml file and parses the pages
+    3 --> Each page is then parsed and passed into KeywordPageDetectorModule for further operations
+    4 --> KeywordPageDetectorModule looks for the keywords and returns the pages in a list called detectedPages
+          back to pageProcessor and stores them in a list called detectedPage
+    '''
+    
+    def run(self, book):
+        self.pageNo = self.pageProcessor(book.xml)
+
+    @staticmethod
+    def pageProcessor(xmlFile):
+        utf8_parser = etree.XMLParser(encoding='utf-8')
+        node = etree.fromstring(xmlFile.encode('utf-8'), parser=utf8_parser)
+        for x in node.iter('OBJECT'):
+            param = x[0].attrib['value'].split('.')[0] 
+            pageNo = param[-4:]
+            detectedPages = KeywordPageDetectorModule.keywordPageFinder(x)
+            print("\n***** Page Processor Results *****\nDetected pages for given keyword are ",detectedPages)
+            return detectedPages
```

Comment:
Rather than `KeyWordPageDetectorModule` we should be using whatever modules are passed into this function.

The whole point of the PageTypeProcessor is we give it modules to be run on every page. :)

```suggestion
    def run(self, book):
        utf8_parser = etree.XMLParser(encoding='utf-8')
        node = etree.fromstring(book.xml.encode('utf-8'), parser=utf8_parser)
        for x in node.iter('OBJECT'):
            param = x[0].attrib['value'].split('.')[0] 
            pageNo = param[-4:]
            for m in self.modules:
                self.modules[m].run(x) 
```